app/validation/validator.py

```diff
@@ -144,6 +144,20 @@ def _validate_questions(self, block_or_variant, numeric_answer_ranges):
 
         return errors
 
+    def _ensure_all_variant_ids_equal(self, block, variants):
```

Comment:
Do we also need to enforce this at an `answer_id` level?

Reply:
I hesitated on this because I think this would also require enforcing the same number of answers across variants? Maybe that use case is something we should use routing for anyway though.

Reply:
I can see a case for different answer options but not different answers. At the point we get to different answers then i think that would not be a variant.

Reply:
Not sure, I can see scenarios where we would have different no. of answers?. If you had situations where a user is either shown a `from date` vs `from date` and `to date`?. 
Or a user being shown a list of checkboxes (1 answer) vs shown a mutually exclusive checkbox (2 answers), these should technically be variants and not really new question blocks?

Enforcing consistent answer ids for also won't work for mutually exclusive either, as the exclusive answer is a diff id..?

Reply:
hmm, in not sure that really fits into the idea of what variants was for. maybe we need to be more explicit on the use cases for variants vs routing?

Reply:
I think its fine as long as we clearly define what is considered a variant vs a new question.

Reply:
Agreed, just spoken to Mebin about this, we need to define the use cases for routing and variants more clearly. The original requirements for variants was to support different versions of:

- question titles
- answer guidance
- interstitials
- detail answer
- response options

Having from_date and to_date seems like a different question to me.